builder/publish.go

```diff
@@ -51,47 +77,88 @@ func PublishImage(image string, handler string, functionName string, language st
 			return nil
 		}
 
-		buildOptPackages, buildPackageErr := getBuildOptionPackages(buildOptions, language, langTemplate.BuildOptions)
+		if remoteBuilder != "" {
+			logger := log.New(os.Stderr, "", log.LstdFlags)
+			logger.SetFlags(log.Ldate | log.Ltime | log.LUTC)
 
-		if buildPackageErr != nil {
-			return buildPackageErr
+			tempDir, err := os.MkdirTemp(os.TempDir(), "builder-*")
+			if err != nil {
+				return fmt.Errorf("failed to create temporary directory for %s, error: %w", functionName, err)
+			}
+			defer os.RemoveAll(tempDir)
 
-		}
+			tarPath := path.Join(tempDir, "req.tar")
 
-		dockerBuildVal := dockerBuild{
-			Image:            imageName,
-			NoCache:          nocache,
-			Squash:           squash,
-			HTTPProxy:        os.Getenv("http_proxy"),
-			HTTPSProxy:       os.Getenv("https_proxy"),
-			BuildArgMap:      buildArgMap,
-			BuildOptPackages: buildOptPackages,
-			BuildLabelMap:    buildLabelMap,
-			Platforms:        platforms,
-			ExtraTags:        extraTags,
-		}
+			if err := makeTar(buildConfig{Image: imageName}, path.Join("build", functionName), tarPath); err != nil {
+				return fmt.Errorf("failed to create tar file for %s, error: %w", functionName, err)
+			}
 
-		command, args := getDockerBuildxCommand(dockerBuildVal)
-		fmt.Printf("Publishing with command: %v %v\n", command, args)
+			res, err := callBuilder(logger, tarPath, tempPath, remoteBuilder, functionName, payloadSecretPath)
+			if err != nil {
+				return err
+			}
+			defer res.Body.Close()
 
-		task := v1execute.ExecTask{
-			Cwd:         tempPath,
-			Command:     command,
-			Args:        args,
-			StreamStdio: !quietBuild,
-		}
+			data, _ := io.ReadAll(res.Body)
 
-		res, err := task.Execute()
+			result := buildResult{}
+			if err := json.Unmarshal(data, &result); err != nil {
+				return err
+			}
 
-		if err != nil {
-			return err
-		}
+			for _, logMsg := range result.Log {
+				fmt.Printf("%s\n", logMsg)
+			}
```

Comment:
@alexellis Is this what you expect for the build logs? I have updated the PR description with the logs.

Reply:
Yes something like that 👍 

basically, this information is shown unless --quiet is passed. In which case you shouldn't display it (like with the regular faas-cli build/publish command

Reply:
Added condition for printing logs.